src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+
+        [TestMethod]
+        public async Task Dial_Peer_No_Address()
+        {
+            var peer = new Peer
+            {
+                Id = mars.PeerId
+            };
+
+            var swarm = new SwarmService {LocalPeer = self};
+            await swarm.StartAsync();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() => { swarm.DialAsync(peer, "/foo/0.42.0").Wait(); });
+            }
+            finally
+            {
+                await swarm.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task Dial_Peer_Not_Listening()
+        {
+            var peer = new Peer
+            {
+                Id = mars.PeerId,
+                Addresses = new List<MultiAddress>
+                {
+                    new MultiAddress($"/ip4/127.0.0.1/tcp/4242/ipfs/{mars.PeerId}"),
+                    new MultiAddress($"/ip4/127.0.0.2/tcp/4242/ipfs/{mars.PeerId}")
+                }
+            };
+
+            var swarm = new SwarmService {LocalPeer = self};
+            await swarm.StartAsync();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() => { swarm.DialAsync(peer, "/foo/0.42.0").Wait(); });
+            }
+            finally
+            {
+                await swarm.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task Dial_Peer_UnknownProtocol()
+        {
+            var peerB = new Peer
+            {
+                AgentVersion = "peerB",
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                PublicKey =
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            var peerBAddress = await swarmB.StartListeningAsync("/ip4/127.0.0.1/tcp/0");
+
+            var swarm = new SwarmService {LocalPeer = self};
+            await swarm.StartAsync();
+            try
+            {
+                ExceptionAssert.Throws<Exception>(() => { swarm.DialAsync(peerB, "/foo/0.42.0").Wait(); });
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task Dial_Peer()
+        {
+            var peerB = new Peer
+            {
+                AgentVersion = "peerB",
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                PublicKey =
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            await swarmB.StartAsync();
+            var peerBAddress = await swarmB.StartListeningAsync("/ip4/127.0.0.1/tcp/0");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'peerBAddress'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)